conjure-java-core/src/main/java/com/palantir/conjure/java/types/BeanBuilderGenerator.java

```diff
@@ -249,6 +254,9 @@ private MethodSpec createSetter(EnrichedField enriched) {
                 .addMember("value", "$S", enriched.fieldName().get());
         if (isCollectionType(type)) {
             annotationBuilder.addMember("nulls", "$T.SKIP", Nulls.class);
+            if (featureFlags.contains(FeatureFlags.NonNullCollections) && !isOptionalInnerType(type)) {
```

Comment:
A couple more edge cases (sorry!):
- I think inner type of `any` should be handled the same way as optional, optional.empty can go in, null is deserialized. Ideally we'd deserialize null (object) to empty optional, but I don't think jackson has a flag for that.
- Do we need to dealias `type`? I don't recall if `list<alias<optional<?>>>` is deserialized quite properly, if memory serves we get a null alias object instead of an alias of an empty optional.

Reply:
- The [spec](https://github.com/palantir/conjure/blob/master/docs/spec/wire.md#51-built-in-types) says any is a primitive or an object with any fields which to me would suggest it be a non-nulllable value so I think we want to keep the current behaviour for any. 

We currently do not correctly support aliases of optionals. I tried playing around with different flags to get it to work, but i wasn't successful. I'll push an additional test demonstrating to current behaviour

Reply:
That's fair, any time an optional value is used for `any`, we should recommend using `optional<any>` instead. I got pushback on that from a product using the `any` type for backcompat where they didn't want to change the types in generated bindings for feat of breaking consumers. In that case they can defer adding the NonNullCollections flag until they've made the conjure definition change.